File: src/lessons/welcome/array/search/OccurrenceOfValueEntity.java
package lessons.welcome.array.search;

import static plm.core.ValueSerializer.*;

import plm.universe.bat.BatEntity;

public class OccurrenceOfValueEntity extends BatEntity {

  public void run()
  {
    int count = getTestCount();
    for (int i = 0; i < count; i++) {
      Object[] param = (Object[])deserialize(getTest(i));
      setTestResult(i, serialize(occurrences(toPrimitive(toArrayOfType(param[0], Integer.class)), (Integer)param[1])));
    }
  }

  /* BEGIN TEMPLATE */
  // counts the occurrences of the value 'lookingFor' contained in tab variable
  public int occurrences(int[] tab, int lookingFor)
  {
    /* BEGIN SOLUTION */
    int count = 0;
    for (int i = 0; i < tab.length; i++) {
      if (tab[i] == lookingFor) {
        count++;
      }
    }
    return count;
    /* END SOLUTION */
  }

  /* END TEMPLATE */
}

File: src/lessons/welcome/array/search/OccurrenceOfValueEntity.py
def run():
    count = getTestCount()
    for i in range(count):
        param = deserialize(getTest(i))
        setTestResult(i, serialize(occurrences(param[0], param[1])))

# BEGIN TEMPLATE
def occurrences(nums,lookingFor):
# BEGIN SOLUTION
  count = 0
  for i in range(len(nums)):
    if nums[i] == lookingFor:
      count += 1
  return count
# END SOLUTION
# END TEMPLATE
